packages/components/mention/src/mention.vue

```diff
@@ -0,0 +1,247 @@
+      dropdownRef.value?.selectHoverOption()
+    } else {
+      visible.value = false
+    }
+  } else if (['Backspace'].includes(e.key)) {
+    if (props.whole && mentionCtx.value) {
+      const { splitIndex, selectionEnd, pattern, prefixIndex, prefix } =
+        mentionCtx.value
+      const inputEl = getInputEl()
+      if (!inputEl) return
+      const inputValue = inputEl.value
+      const matchOption = props.options.find((item) => item.value === pattern)
+      const isWhole = isFunction(props.checkIsWhole)
+        ? props.checkIsWhole(pattern, prefix)
+        : matchOption
+      if (isWhole && splitIndex !== -1 && splitIndex + 1 === selectionEnd) {
+        e.preventDefault()
+        const newValue =
+          inputValue.slice(0, prefixIndex) + inputValue.slice(splitIndex + 1)
+        emit(UPDATE_MODEL_EVENT, newValue)
+
+        const newSelectionEnd = prefixIndex
+        nextTick(() => {
+          // input value is updated
+          inputEl.selectionStart = newSelectionEnd
+          inputEl.selectionEnd = newSelectionEnd
+          syncDropdownVisible()
+        })
+      }
+    }
+  }
+}
+
+const { wrapperRef, handleFocus, handleBlur } = useFocusController(elInputRef, {
+  afterFocus() {
+    syncAfterCursorMove()
+  },
+  beforeBlur(event) {
+    return tooltipRef.value?.isFocusInsideContent(event)
+  },
+  afterBlur() {
+    visible.value = false
+  },
+})
+
+const handleInputMouseDown = () => {
+  syncAfterCursorMove()
+}
+
+const handleSelect = (item: MentionOption) => {
+  if (!mentionCtx.value) return
+  const inputEl = getInputEl()
+  if (!inputEl) return
+  const inputValue = inputEl.value
+  const { split } = props
+
+  const newEndPart = inputValue.slice(mentionCtx.value.end)
+  const alreadySeparated = newEndPart.startsWith(split)
+  const newMiddlePart = `${item.value}${alreadySeparated ? '' : split}`
+
+  const newValue =
+    inputValue.slice(0, mentionCtx.value.start) + newMiddlePart + newEndPart
+
+  emit(UPDATE_MODEL_EVENT, newValue)
+  emit('select', item, mentionCtx.value.prefix)
+
+  const newSelectionEnd =
+    mentionCtx.value.start + newMiddlePart.length + (alreadySeparated ? 1 : 0)
+
+  nextTick(() => {
+    // input value is updated
+    inputEl.selectionStart = newSelectionEnd
+    inputEl.selectionEnd = newSelectionEnd
+    syncDropdownVisible()
+  })
+}
+
+const getInputEl = () =>
+  props.type === 'textarea'
+    ? elInputRef.value?.textarea
+    : elInputRef.value?.input
+
+const syncAfterCursorMove = () => {
+  // can't use nextTick(), get cursor position will be wrong
+  setTimeout(() => {
+    syncCursor()
+    syncDropdownVisible()
+    nextTick(() => tooltipRef.value?.updatePopper())
+  }, 0)
+}
+
+const syncCursor = () => {
+  const inputEl = getInputEl()
+  if (!inputEl) return
+
+  const caretPosition = getCursorPosition(inputEl)
+  cursorStyle.value = {
+    position: 'absolute',
+    width: 0,
+    height: 0,
+    left: `${caretPosition.left + inputEl.offsetLeft}px`,
```

Comment:
IMO, the left needs to be obtained based on getBoundingClientRect to avoid misalignment. [Bug](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtbWVudGlvblxuICAgIHYtbW9kZWw9XCJ2YWx1ZVwiXG4gICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICBzdHlsZT1cIndpZHRoOiAzMjBweFwiXG4gICAgcGxhY2Vob2xkZXI9XCJQbGVhc2UgaW5wdXRcIlxuICAvPlxuICA8ZWwtbWVudGlvblxuICAgIHYtbW9kZWw9XCJ2YWx1ZVwiXG4gICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICBzdHlsZT1cIndpZHRoOiAzMjBweFwiXG4gICAgcGxhY2Vob2xkZXI9XCJQbGVhc2UgaW5wdXRcIlxuICA+XG4gICAgPHRlbXBsYXRlICNwcmVwZW5kPm1pc2FsaWdubWVudDwvdGVtcGxhdGU+XG4gIDwvZWwtbWVudGlvbj5cbiAgPGRpdiBzdHlsZT1cImRpc3BsYXk6IGZsZXhcIj5cbiAgICA8ZWwtbWVudGlvblxuICAgICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICAgIDpvcHRpb25zPVwib3B0aW9uc1wiXG4gICAgICBzdHlsZT1cIndpZHRoOiAzMjBweFwiXG4gICAgICBwbGFjZWhvbGRlcj1cIlBsZWFzZSBpbnB1dFwiXG4gICAgLz5cbiAgICA8ZWwtbWVudGlvblxuICAgICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICAgIDpvcHRpb25zPVwib3B0aW9uc1wiXG4gICAgICBzdHlsZT1cIndpZHRoOiAzMjBweFwiXG4gICAgICBwbGFjZWhvbGRlcj1cIlBsZWFzZSBpbnB1dFwiXG4gICAgPlxuICAgICAgPHRlbXBsYXRlICNwcmVwZW5kPm1pc2FsaWdubWVudDwvdGVtcGxhdGU+XG4gICAgPC9lbC1tZW50aW9uPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5jb25zdCB2YWx1ZSA9IHJlZignQCcpXG5cbmNvbnN0IG9wdGlvbnMgPSByZWYoW1xuICB7XG4gICAgbGFiZWw6ICdGdXBob2VuaXhlcycsXG4gICAgdmFsdWU6ICdGdXBob2VuaXhlcycsXG4gIH0sXG4gIHtcbiAgICBsYWJlbDogJ2tvb3Jpb29rYW1pJyxcbiAgICB2YWx1ZTogJ2tvb3Jpb29rYW1pJyxcbiAgfSxcbiAge1xuICAgIGxhYmVsOiAnSmVyZW15JyxcbiAgICB2YWx1ZTogJ0plcmVteScsXG4gIH0sXG4gIHtcbiAgICBsYWJlbDogJ2J0ZWEnLFxuICAgIHZhbHVlOiAnYnRlYScsXG4gIH0sXG5dKVxuPC9zY3JpcHQ+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMTc1ODYtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTE3NTg2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGwoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE3NTg2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xNzU4Ni1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

```ts
caretPosition.left - elInputRef.value!.$el.getBoundingClientRect().left + inputEl.getBoundingClientRect().left
```

Reply:
thanks